core-api/src/main/java/com/optimizely/ab/config/Experiment.java

```diff
@@ -173,6 +173,101 @@ public boolean isLaunched() {
         return status.equals(ExperimentStatus.LAUNCHED.toString());
     }
 
+    public String serializeConditions(Map<String, String> audiencesMap) {
+        Condition condition = this.audienceConditions;
+        return condition instanceof EmptyCondition ? "" : this.serialize(condition, audiencesMap);
+    }
+
+    private String getNameFromAudienceId(String audienceId, Map<String, String> audiencesMap) {
+        if (audiencesMap == null ) return "\"" + audienceId + "\"";
```

Comment:
nit: can we use format or stringbuilder in concatenation?

Reply:
this applies to all if you agree.

Reply:
Since there is no loop in this function, the compiler will actually use StringBuilder in the background. However we can explicitly use here as well.